core/src/main/java/hudson/model/Computer.java

```diff
@@ -1175,6 +1176,32 @@ public String getSearchUrl() {
         return getEnvironment();
     }
 
+    /**
+     * Expose real full env vars map from agent for UI presentation
+     */
+    public Map<String,String> getEnvVarsFull() throws IOException, InterruptedException {
+        if(getChannel() == null) {
+            Map<String, String> env = new TreeMap<> ();
+            env.put("N/A","N/A");
+            return env;
+        } else {
+            return getChannel().call(new ListFullEnvironment());
+        }
+    }
+
+    private static class ListFullEnvironment extends MasterToSlaveCallable<Map<String,String>,IOException> {
+        public Map<String,String> call() throws IOException {
+            Map<String, String> env = new TreeMap(System.getenv());
```

Comment:
```suggestion
            Map<String, String> env = new TreeMap<>(System.getenv());
```

To avoid an "unchecked assignment"